### ISSUE TITLE
incorrect node numbering in 'flux resource list'

### ISSUE THREAD

**Issue by ryanday36** [flux-framework/flux-core#4178]
On corona, we have three login nodes (corona[82,211-212]) with 24 cores each, whereas the compute nodes have 48 cores each. In this scenario, the node numbering in the 'NODELIST' column of the `flux resource list` output is incorrect, but in different ways depending on how the nodes are listed in `bootstrap.toml`. 

In order to work around an issue with `flux ion-R encode` (flux-sched issue 912), the login nodes are listed as the last ranks in R:
```
"version": 1, "execution": {"R_lite": [{"rank": "0-164", "children": {"core": "0-47"}}, {"rank": "165-167", "children": {"core": "0-23"}}], "starttime": 0.0, "expiration": 0.0, "nodelist": ["corona[81,1-80,87-170,82,211-212]"]}
```
If I list the login nodes last in `bootstrap.toml` to match `R`, then the node numbering for the 'free' nodes in `flux resource list` is off by three, but the node numbers in the `down` list appear to be correct. If I list the login nodes before the compute nodes in `bootstrap.toml`, the 'free' nodes have the correct numbering, but the 'down' nodes are off by three. E.g.

login nodes listed last in `bootstrap.toml` ('down' nodes have the expected hostlist):
```
[day36@corona81:flux]$ flux resource list
     STATE NNODES   NCORES    NGPUS NODELIST
      free     35     1680        0 corona[5,10,12-13,19,22,33,42-43,47,49,65-66,69,71,73,77,95,106,111-113,122,124,129-130,132,136,138-139,146,151-153,157]
 allocated      0        0        0 
      down    133     6312        0 corona[82,211-212,81,1-7,9-12,14,17-21,23-24,26-35,37-44,47-49,51,53-67,70-71,73,75,77-79,87-97,99-108,110-113,117-124,126,128-131,134,136-138,140,143-148,150-153,157-159,161-170]
[day36@corona81:flux]$
```
login nodes list before compute nodes in `bootstrap.toml` ('free' nodes have the expected hostlist):
```
[day36@corona81:flux]$ flux resource list
     STATE NNODES   NCORES    NGPUS NODELIST
      free     35     1680        0 corona[8,13,15-16,22,25,36,45-46,50,52,68-69,72,74,76,80,98,109,114-116,125,127,132-133,135,139,141-142,149,154-156,160]
 allocated      0        0        0 
      down    133     6312        0 corona[82,211-212,81,1-10,12-15,17,20-24,26-27,29-38,40-47,50-52,54,56-70,73-74,76,78,80,87-88,90-100,102-111,113-116,120-127,129,131-134,137,139-141,143,146-151,153-156,160-162,164-170]
[day36@corona81:flux]$
```

and, for reference, the 'hosts' section of `bootstrap.toml` with the login nodes listed last:
```
hosts = [
{ host = "corona81", bind = "tcp://192.168.64.81:%p", connect = "tcp://ecorona81:%p" },
{ host = "corona[1-80,87-170,82,211-212]" },

]
```

**Comment by grondo** [comment]
The result of `flux resource list` is provided by the scheduler, so my initial guess in that this will be fixed by flux-framework/flux-sched#893

**Comment by grondo** [comment]
This should now be fixed with the combination of at least flux-core v0.37.0 and flux-sched v0.21.1.
